cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataEntry.java

```diff
@@ -24,25 +24,25 @@
  * Represents a single Metadata entry for a CDAP Entity.
  */
 public class MetadataEntry {
-  private final Id.NamespacedId targetId;
+  private final NamespacedId targetId;
```

Comment:
even this is persisted to the metadata store. @poornachandra how about backward compatibility here?